File: modified_scripts/julia/bubble_sort/gemma_7b.jl

function bubble_sort(arr)
  for i in 1:length(arr) - 1
    for j in 1:length(arr) - i
      if arr[j] > arr[j + 1]
        arr[j], arr[j + 1] = arr[j + 1], arr[j]
      end
    end
  end

  arr
end

File: modified_scripts/scala/bubble_sort/gemma_7b.scala

object BubbleSort {

  def sort(arr: Array[Int]): Array[Int] = {
    var j = arr.length
    while (j > 1) {
      j = arr.length
      for (i <- 0 until j - 1) {
        if (arr(i) > arr(i + 1)) {
          arr(i) = arr(i) ^ arr(i + 1)
          arr(i + 1) = arr(i) ^ arr(i + 1)
          arr(i) = arr(i) ^ arr(i + 1)
        }
      }
    }
    arr
  }

}
